### ISSUE TITLE
Duplicating pixels, DataLoader, pooling functions

### ISSUE THREAD

**Issue by LouisStef** [VSainteuf/pytorch-psetae#19]
**Observations**
- Because the DataLoader expects samples of same shape (N, T, C, S), you duplicate pixels, such that all samples have the same S
- You save a mask of original/duplicated pixels for the aggregation layer, used by custom pooling functions.
- Drawback : forward passes are applied on duplicated pixels, but the results are thrown away during aggregation = wasted ressources and a more complex pipeline

**Question**
Did you consider any other workaround to handle variable S value between samples ? 
Did you consider any solution which doesn't require building a mask ? (torch.geometric DataLoader for instance ?)

**Comment by VSainteuf** [comment]
Hi again, 
I remember I once did a quick test with PyTorch Scatter to handle that more elegantly. But I didn't push much further because I didn't see noticeable gains. I think that the amount of unnecessary computation is quite limited.

**Comment by LouisStef** [comment]
I am trying to remove the mask to connect the whole model to explainability frameworks (Shap, Captum ...), which usually expect a list of tensors as inputs (our case) but all tensors are interpreted as features (the mask is not one)

I think Torch geometric's DataLoader is an promising candidate, bc it can create batches of graphes which possibly contain different number of nodes ... (A pixel set is a graph without edges)

Will see if I find something interesting !
